.github/workflows/next-sync.yml

```diff
@@ -0,0 +1,28 @@
+name: next-sync
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  next-sync:
+    runs-on: ubuntu-latest
+    if: "! contains(toJSON(github.event.commits.*.message), '[skip ci]')"
```

Comment:
Shouldn't this actions run anytime there's a new commit in `main`? No matter if it's marked as `[skip ci]` or not.

Reply:
Hmm, I'm not sure because the [skip ci] commits will include package.json changes, main changelog, something that we must discard when merging to next. So it's not relevant the main changes into next.

Reply:
Hmm, I thought that `next` should be always the same or ahead of `main`. What happens e.g after we release the new version, the `main` will have version `4.0.0`, but `next` will still have version `3.1.4`. Wouldn't it cause troubles down the line?

Ah, I think I got it. Since we have a branching model similar to [git flow](https://nvie.com/posts/a-successful-git-branching-model/), I recall that by default the main branch isn't merged back to develop (similar to what  this action would be). However, it works only because hotfixes (which are direclty merged to `main`) are merged back to `develop`.

I usually use the `git-flow-avh` extension, which automatically merges `main` to `develop` anytime there's a new change. That way I can run `git describe` and see how far ahead of `main` I'm.

Reply:
Yes, I usually worked with git flow strategy that's why I introduced that.